elip-deterministic-blinding-key.mediawiki

```diff
@@ -0,0 +1,84 @@
+<pre>
+  ELIP: ????
+  Layer: Wallet
+  Title: Deterministic Descriptor Blinding Keys
+  Author: Leonardo Comandini <leonardo@blockstream.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/ElementsProject/elips/wiki/Comments:ELIP-????
+  Status: Draft
+  Type: Standards Track
+  Created: 2023-11-08
+  License: BSD-3-Clause
+</pre>
+
+==Introduction==
+
+===Abstract===
+
+This document proposes a standard way to deterministically derive the descriptor blinding key in view descriptors.
+
+===Copyright===
+
+This document is licensed under the 3-clause BSD license.
+
+===Motivation===
+
+CT descriptors as defined in [https://github.com/ElementsProject/ELIPs/blob/main/elip-0150.mediawiki ELIP-150] consist in a descriptor blinding key and a ordinary descriptor.
+
+The descriptor blinding key can be chosen arbitrarily.
+This however increase the amount of data that the user needs to back up.
+
+Otherwise they can be deterministically derived from a BIP32 seed following [https://github.com/satoshilabs/slips/blob/master/slip-0077.md SLIP-77].
+This works for singlesig wallets,
+since the only backup that the user needs is the BIP32 seed (or BIP39 mnemonic).
+However this does not apply to multisig,
+which might have multiple xpubs,
+and thus multiple BIP32 seeds from which the descriptor blinding key can be derived.
+SLIP77 also does not work well when there are multiple BIP44 accounts,
+as they will have the same master blinding key.
+
+This document defines a standard way to deterministically derive the descriptor blinding key from the oridinary descriptor.
+
+==Design==
+
+===Overview===
+We propose a new possible value for the descriptor blinding key
+
+<pre>
+ct(elipXXXX,<DESCRIPTOR>)
+</pre>
+
+Which means that descriptor blinding key to be used is a '''view key''' deterministically derived from the ordinary descriptor <code>DESCRIPTOR</code>
+
+===Drawbacks===
+If a wallet is using this ELIP, anyone knowing the ordinary descriptor will be able to unblind all the corresponding outputs. 
+
+Anyway some users might find this trade off acceptable.
+
+===Specification===
+The following descriptor
+
+<pre>
+ct(elipXXXX,<DESCRIPTOR>)
+</pre>
+
+Is equivalent to
+
+<pre>
+ct(<KEY>,<DESCRIPTOR>)
+</pre>
+
+Where <code>KEY</code> is a view blinding key derived as follow:
+* If <code>DESCRIPTOR</code> is multipath, get the first descriptor
```

Comment:
I would consider concatenating every scriptPubKey coming from the multipath expansion to avoid collisions.

Reply:
IMHO it's not worth it, added a footnote with a detailed rationale

Reply:
I think we ought to concatenate them, since there will always be a small finite number. (I know it's extra code but it's not *that* bad, and closes a theoretical complaint.)

In any case, I'm not really convinced by the stated rationale, which is basically "I'm not aware of anybody using this in a way that'll cause problems". But I do think it's true that, in practice, users are never going to have multiple descriptors that are identical except for having different multi-path branches.

Reply:
Changed to use the concatenated scriptpubkeys